src/seq/nucleotide.jl

```diff
@@ -563,6 +639,32 @@ end
 getindex{T}(seq::NucleotideSequence{T}, r::UnitRange) = NucleotideSequence{T}(seq, r)
 
 
+function setindex!{T}(seq::NucleotideSequence{T}, nt::T, i::Integer)
+    if !seq.mutable
+        error("Cannot mutate an immutable sequence. Call `mutable!(seq)` first.")
+    end
+    if i < 1 || i > length(seq)
+        throw(BoundsError())
```

Comment:
Since we perform a bound checking here, shouldn't we use `@inbounds` in the rest of this function? Or isn't it necessary?